lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -709,17 +709,37 @@ def is_privilege?(user_or_attribute, operation, object_or_attribute)
       end
     end
 
+    # Returns true if the user has the operation on the object, and that operation is granted by
+    # the user attribute scope or its descendants
+    def is_privilege_via_attribute?(user_or_attribute, operation, object_or_attribute, user_attribute_scope)
+      policy_classes_containing_object = policy_classes_for_object_attribute(object_or_attribute)
+      operation_id = operation.try(:unique_identifier) || operation.to_s
+
+      if policy_classes_containing_object.size < 2
+        !accessible_operations(user_or_attribute, object_or_attribute, operation_id, user_attribute_scope).empty?
+      else
+        policy_classes_containing_object.all? do |policy_class|
```

Comment:
could probably clean up the `arr.all? { |el| !el.empty?}` with something like `arr.any? { |el| el.empty?`, but it doesn't look like policy class is actually used in the block.  Did we add tests for multiple policy classes?